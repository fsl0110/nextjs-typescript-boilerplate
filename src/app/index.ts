export * from '../tests/utils/testingLibraryUtils';
