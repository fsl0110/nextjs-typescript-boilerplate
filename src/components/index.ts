// this kind of reexporting may cause problems.
// see: https://nextjs.org/docs/messages/export-all-in-page

export * from './Button/Button';
export * from './Header/Header';
export * from './FormErrors/FormErrors';
