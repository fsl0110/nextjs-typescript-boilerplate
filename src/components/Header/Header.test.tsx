import { render } from 'app';
import { Header } from 'components';

describe('Header', () => {
  it('renders Header component', () => {
    render(
      <Header
        onLogin={() => null}
        onLogout={() => null}
        onCreateAccount={() => null}
      />,
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <Header
        onLogin={() => null}
        onLogout={() => null}
        onCreateAccount={() => null}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
