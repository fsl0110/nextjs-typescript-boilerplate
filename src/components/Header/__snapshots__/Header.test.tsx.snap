// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="flex justify-between items-center px-4 h-16 bg-gray-200"
  >
    <div
      class="text-2xl font-bold"
    >
      <a
        href="/"
      >
        Next.js Boilerplate
      </a>
    </div>
    <div>
      <select>
        <option
          value="en"
        >
          english
        </option>
        <option
          value="de"
        >
          german
        </option>
      </select>
    </div>
  </header>
</DocumentFragment>
`;
